csharp/src/Google.Protobuf/Collections/RepeatedField.cs

```diff
@@ -101,10 +101,11 @@ public void AddEntriesFrom(CodedInputStream input, FieldCodec<T> codec)
                 int length = input.ReadLength();
                 if (length > 0)
                 {
+                    EnsureSize(count + length);
```

Comment:
length doesn't mean the number of elements, but the total byte length of these elements. Depending on the field type, the number of elements can vary from between length/10 (a varint takes at most 10 bytes) to length. I believe this call here will allocate more memory than it needs to be most of the time (double or triple at least).